core/src/main/scala/mongo4cats/bson/BsonDecoder.scala

```diff
@@ -66,6 +79,142 @@ object BsonDecoder {
   implicit val bsonDecoder: BsonDecoder[BsonValue] =
     instance[BsonValue](Right(_))
 
+  implicit val int: BsonDecoder[Int] = BsonDecoder.instance {
+    case x: BsonInt32 => Right(x.getValue)
+    case _            => Left(BsonDecodeError("Expected BsonInt32"))
+  }
```

Comment:
Do we want to avoid stack trace from one decoder calling another, or why not reuse logic (e.g. define decoders for `Bson*` types and then define Int, Long, etc with mapping over previous ones)?

Reply:
Ah yeah, now I've added `flatMap` reusing the other decoders makes sense, since that's exactly what I added it for :smile: